Cabal/src/Distribution/PackageDescription/Check/Conditional.hs

```diff
@@ -107,12 +108,55 @@ annotateCondTree fs ta (CondNode a c bs) =
           )
           fs
 
+    defTrueFlags :: [PackageFlag]
+    defTrueFlags = filter flagDefault fs
+
+-- Propagate contextual information in CondTree branches. This is
+-- needed as CondTree is a rosetree and not a binary tree.
+crossAnnotateBranches
+  :: forall a
+   . (Eq a, Monoid a)
+  => [PackageFlag] -- `default: true` flags.
+  -> [CondBranch ConfVar [Dependency] (TargetAnnotation a)]
+  -> [CondBranch ConfVar [Dependency] (TargetAnnotation a)]
+crossAnnotateBranches fs bs = map crossAnnBranch bs
+  where
+    crossAnnBranch
+      :: CondBranch ConfVar [Dependency] (TargetAnnotation a)
+      -> CondBranch ConfVar [Dependency] (TargetAnnotation a)
+    crossAnnBranch wr =
+      let
+        rs = filter (/= wr) bs
+        ts = mapMaybe realiseBranch rs
+       in
+        utaBranch (mconcat ts) wr
+
+    realiseBranch :: CondBranch ConfVar [Dependency] (TargetAnnotation a) -> Maybe a
+    realiseBranch b =
+      let
+        -- We are only interested in True by default package flags.
+        rbf :: ConfVar -> Either ConfVar Bool
+        rbf (PackageFlag n) | elem n (map flagName fs) = Right True
+        rbf _ = Right False
+        ms = simplifyCondBranch rbf (fmap taTarget b)
+       in
+        fmap snd ms
+
+    utaBranch
+      :: a
+      -> CondBranch ConfVar [Dependency] (TargetAnnotation a)
+      -> CondBranch ConfVar [Dependency] (TargetAnnotation a)
+    utaBranch a (CondBranch k t mt) =
+      let utt (CondNode ka c wbs) =
+            (CondNode (updateTargetAnnotation a ka) c wbs)
+       in CondBranch k (utt t) (utt <$> mt)
```

Comment:
I'm not familiar with this part of the codebase. What do `rbf`,  `utt` and `uta` mean?

Reply:
realiseBranchFunction, updateTargetAnnotation, updateTargetTree.

There is actually no need to have them as across, I will convert them to longhand.